stream_buffer.c

```diff
@@ -724,51 +724,40 @@ static size_t prvWriteMessageToBuffer( StreamBuffer_t * const pxStreamBuffer,
                                        size_t xSpace,
                                        size_t xRequiredSpace )
 {
-    BaseType_t xShouldWrite;
-    size_t xReturn;
     size_t xNextHead = pxStreamBuffer->xHead;
 
-    if( xSpace == ( size_t ) 0 )
+    if( xDataLengthBytes != xRequiredSpace )

```

Comment:
Could alternatively rewrite this line as:
```c
if( ( pxStreamBuffer->ucFlags & sbFLAGS_IS_MESSAGE_BUFFER ) != ( uint8_t ) 0 )
```
They're currently equivalent. The noisier version might be clearer or safer long-term.